src/ooni/utils.hpp

```diff
@@ -5,34 +5,86 @@
 #ifndef SRC_OONI_UTILS_HPP
 #define SRC_OONI_UTILS_HPP
 
+#include "src/ext/json/src/json.hpp"
+#include <GeoIP.h>
 #include <measurement_kit/common.hpp>
 #include <measurement_kit/http.hpp>
 #include <regex>
 #include <string>
 
+using Json = nlohmann::json;
+
 namespace mk {
 namespace ooni {
 
 template <decltype(mk::http::get) httpget = mk::http::get>
 void ip_lookup(Callback<std::string> callback) {
     httpget("http://geoip.ubuntu.com/lookup",
-              [=](Error err, http::Response response) {
-                  if (err) {
-                      callback(err, "");
-                      return;
-                  }
-                  if (response.status_code != 200) {
-                      callback(GenericError(), "");
-                      return;
-                  }
-                  std::smatch m;
-                  std::regex regex("<Ip>(.*)</Ip>");
-                  if (std::regex_search(response.body, m, regex) == false) {
-                      callback(GenericError(), "");
-                      return;
-                  }
-                  callback(NoError(), m[1]);
-              }, {}, "", {}, Logger::global(), Poller::global());
+            [=](Error err, http::Response response) {
+                if (err) {
+                    callback(err, "");
+                    return;
+                }
+                if (response.status_code != 200) {
+                    callback(GenericError(), "");
+                    return;
+                }
+                std::smatch m;
+                std::regex regex("<Ip>(.*)</Ip>");
+                if (std::regex_search(response.body, m, regex) == false) {
+                    callback(GenericError(), "");
+                    return;
+                }
+                callback(NoError(), m[1]);
+            },
+            {}, "", {}, Logger::global(), Poller::global());
+}
+
+// this is needed to return a const char instead of a char for asn resolver
+inline GEOIP_API const char *
+MK_GeoIP_name_by_name_gl(GeoIP *gi, const char *host, GeoIPLookup *gl) {
+    return (const char *)GeoIP_name_by_name_gl(gi, host, gl);
+}
+
+template <decltype(GeoIP_country_code3_by_name_gl) resolver>
+std::string geoip_query(std::string ip, std::string path) {
+    GeoIP *gi;
+    GeoIPLookup gl;
```

Comment:
GeoIPLookup is [a struct defined in GeoIP.h](https://github.com/maxmind/geoip-api-c/blob/master/libGeoIP/GeoIP.h#L80). We need it because we are using the new api to do queries (because the old [are deprecated](https://github.com/maxmind/geoip-api-c/blob/master/libGeoIP/GeoIP.h#L378)), but I was not able to find examples for the new apis. 

The struct GeoIPLookup contains an integer that should be the netmask. I don't know actually if it is just an out variable or also an in-out variable, and I was not able to find documentation for this. @bassosimone